src/core/dev/ring_simple.cpp

```diff
@@ -407,66 +407,51 @@ void ring_simple::create_resources()
     ring_logdbg("new ring_simple() completed");
 }
 
-int ring_simple::request_notification(cq_type_t cq_type, uint64_t poll_sn)
+bool ring_simple::request_notification(cq_type_t cq_type)
 {
-    if (!safe_mce_sys().doca_flow) {
-        int ret = 1;
-        if (likely(CQT_RX == cq_type)) {
-            RING_TRY_LOCK_RUN_AND_UPDATE_RET(m_lock_ring_rx,
-                                             m_p_cq_mgr_rx->request_notification(poll_sn);
-                                             ++m_p_ring_stat->simple.n_rx_interrupt_requests);
-        } else {
-            RING_TRY_LOCK_RUN_AND_UPDATE_RET(m_lock_ring_tx,
-                                             m_p_cq_mgr_tx->request_notification(poll_sn));
-        }
-        return ret;
-    }
-
     bool ret = false;
     if (likely(CQT_RX == cq_type)) {
         m_lock_ring_rx.lock();
-        ret = m_hqrx->request_notification();
+        if (!safe_mce_sys().doca_flow) {
+            ret = m_p_cq_mgr_rx->request_notification();
+        } else {
+            ret = m_hqrx->request_notification();
+        }
         ++m_p_ring_stat->simple.n_rx_interrupt_requests;
         m_lock_ring_rx.unlock();
     } else {
         m_lock_ring_tx.lock();
-        ret = (m_p_cq_mgr_tx->request_notification(poll_sn) >= 0);
+        ret = m_p_cq_mgr_tx->request_notification();
         m_lock_ring_tx.unlock();
     }
 
-    return (ret ? 0 : -1);
+    return ret;
 }
 
-int ring_simple::wait_for_notification_and_process_element(int cq_channel_fd,
-                                                           uint64_t *p_cq_poll_sn,
-                                                           void *pv_fd_ready_array /*NULL*/)
+void ring_simple::clear_rx_notification()
 {
     if (!safe_mce_sys().doca_flow) {
-        int ret = -1;
+        m_lock_ring_rx.lock();
```

Comment:
[suggestion] replace manual locking with a lock_guard at the top

Reply:
Done